src/DiagonalHessianApproximation.jl

```diff
@@ -63,7 +63,7 @@ function push!(
   sT_B_s = dot(s2, B.d)
   q = sT_y - sT_B_s
   q /= trA2
-  B.d .+= q .* s .^ 2
```

Comment:
Do you understand why this allocates? I used generators on lines 58 and 59 exactly because I thought they would not allocate.

Reply:
It seems like any operation involving a generator is allocating.

```julia
# Consider B.d .= s

julia> @wrappedallocs push!(D, s0, y0)
80
```

I believe that since B.d is a vector, it seems to require allocating a new vector for the generator s to perform the operation.

Reply:
However,
```julia
julia> v = collect(1:10)
julia> u = 1:10
julia> u = (vi^2 for vi in v)
julia> @wrappedallocs w .= v .+ u
0
```